File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode * curr = head;
        ListNode * prev = NULL;
        int len=0;
        while(curr!=NULL)
        {
            len++;
            curr=curr->next;
        }
        curr=head;
        int node = len-n;
        while(node>0)
        {
            prev=curr;
            curr=curr->next;
            node--;
        }
        if(prev) prev->next=curr->next;
        else head=curr->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode curr = head;
        int len = 0;
        while(curr!=null)
        {
            curr=curr.next;
            len++;
        }
        curr=head;
        int node = len - n;
        if(len==n) return head.next;
        while(node>1)
        {
            curr=curr.next;
            node--;
        }
        curr.next=curr.next.next;
        return head;
    }
}